docs/source/_templates/sdk_TP_tutorial.rst

```diff
@@ -0,0 +1,378 @@
+{% set short_lang = 'python' %}
+{% if language == 'JavaScript' %}
+    {% set short_lang = 'js' %}
+{% endif %}
+
+{% set lowercase_lang = 'python' %}
+{% if language == 'JavaScript' %}
+    {% set lowercase_lang = 'javascript' %}
+{% endif %}
+
+***********************************************
+Transaction Processor Tutorial  {{ language }}
+***********************************************
+
+Overview
+========
+
+This tutorial covers the creation of a new Sawtooth Lake transaction family in
+{{ language }}, based on the Sawtooth Lake SDK. We will construct a transaction
+handler which implements a distributed version of the multi-player game tic-
+tac-toe
+
+.. note::
+
+    The SDK contains a fully-implemented version of tic-tac-toe. This tutorial
+    is meant to demonstrate the relevant concepts, rather than to create a
+    complete implementation. See the SDK_ for full implemenations in
+    multiple languages.
+
+.. _SDK: https://github.com/hyperledger/sawtooth-core/tree/master/sdk
+
+A general description of tic-tac-toe, including the rules, can be found on
+Wikipedia at:
+
+    https://en.wikipedia.org/wiki/Tic-tac-toe
+
+A full implementation of the tic-tac-toe transaction family can be found in
+
+``/project/sawtooth-core/sdk/examples/xo_{{ lowercase_lang }}/``.
+
+Prerequisites
+=============
+
+This tutorial assumes that you have gone through :doc:`/app_developers_guide/getting_started` and are
+familiar with the concepts introduced there.
+
+Prior to going through this tutorial, you should have a working vagrant
+environment running to which you can login.  Specific setup instructions are
+available in :doc:`/app_developers_guide/getting_started`.
+
+The Transaction Processor
+=========================
+
+There are two top-level components of a transaction processor: a processor
+class and a handler class. The SDK provides a general-purpose processor class.
+The handler class is application-dependent and contains the business logic for
+a particular family of transactions. Multiple handlers can be connected to a
+processor class.
+
+Handlers get called in two ways: an ``apply`` method and various metadata
+methods. The metadata is used to connect the handler to the processor, and
+we'll discuss it at the end of this tutorial. The bulk of the handler, however,
+is made up of ``apply`` and its helper functions, so that's where we'll start.
+
+The ``apply`` Method
+====================
+
+{% if language == 'JavaScript' %}
+``apply`` gets called with two arguments, ``transactionProcessRequest`` and ``state``.
+``transactionProcessRequest`` holds the command that is to be executed (e.g. taking a space or
+creating a game), while ``state`` stores information about the current
+state of the game (e.g. the board layout and whose turn it is).
+
+{% else %}
+``apply`` gets called with two arguments, ``transaction`` and ``state_store``.
+``transaction`` holds the command that is to be executed (e.g. taking a space or
```

Comment:
Since this is a general transaction handler/family/processor doc, I would mention that the 'transaction' contains payload bytes that are opaque to the validator core, and transaction family specific. When implementing a transaction handler the binary serialization protocol is up to the implementer.